### PR TITLE
Run unit tests on 1.18 in addition to 1.17

### PR DESCRIPTION
**Pull request by dashpole**
**Description:**
Run unit tests on 1.18 in addition to 1.17.

Related to https://github.com/open-telemetry/opentelemetry-collector/issues/5032, but for contrib.

In order to merge this PR, an admin for the repo will need to change the branch protection rules for main to require the `build-and-test / unittest (1.18)` and `build-and-test / unittest (1.17)` checks instead of the `unittest` check.

### CODE REVIEW

**Review by codeboten — APPROVED** [review]

**Comment by dashpole** [comment]
I still see `unittest Expected — Waiting for status to be reported (Required)` in the tests

**Comment by codeboten** [comment]
Sadly even a maintainer can't update branch protection :(. we need someone from the TC.

cc @bogdandrutu

**Comment by tigrannajaryan** [comment]
@bogdandrutu PTAL.

I think we need to disable the unittest from protections temporarily, merge this PR and add the 2 new protections.

**Comment by tigrannajaryan** [comment]
unittest action removed from protections, PR merged, waiting to finish to add the new actions to protections...

**Comment by tigrannajaryan** [comment]
unittest (1.17) and unittest (1.18) action added to protections.

**Comment by tigrannajaryan** [comment]
@dashpole the build failed after merging:
https://github.com/open-telemetry/opentelemetry-collector-contrib/runs/5695569245

**Comment by dashpole** [comment]
> Error: /home/runner/go/pkg/mod/google.golang.org/grpc@v1.45.0/internal/transport/controlbuf.go:30:2: golang.org/x/net@v0.0.0-20220225172249-27dd8689420f: stream error: stream ID 57; INTERNAL_ERROR; received from peer
Error: /home/runner/go/pkg/mod/google.golang.org/grpc@v1.45.0/internal/transport/controlbuf.go:31:2: golang.org/x/net@v0.0.0-20220225172249-27dd8689420f: stream error: stream ID 57; INTERNAL_ERROR; received from peer
Error: /home/runner/go/pkg/mod/google.golang.org/grpc@v1.45.0/server.go:36:2: golang.org/x/net@v0.0.0-20220225172249-27dd8689420f: stream error: stream ID 57; INTERNAL_ERROR; received from peer

Seems like it isn't able to `go get` grpc?

**Comment by tigrannajaryan** [comment]
Probably a transient network error.

**Pull request merged**
